distributed/actor.py

```diff
@@ -118,13 +119,33 @@ def __dir__(self):
         return sorted(o)
 
     def __getattr__(self, key):
-        attr = getattr(self._cls, key)
 
         if self._future and self._future.status not in ("finished", "pending"):
             raise ValueError(
                 "Worker holding Actor was lost.  Status: " + self._future.status
             )
 
+        if (
+            self._worker
+            and self._worker.address == self._address
+            and threading.current_thread().name.startswith("Dask-Actor-Threads")
```

Comment:
Typically we use threadlocals for this.  You might want to look into the use of the `thread_state` variable in worker.py